### ISSUE TITLE
wrong timestamps

### ISSUE THREAD

**Issue by aaannndddyyy** [irungentoo/toxcore#1440]
when a contact writes something at t to you, and you come online at t1=t+Delta, then the message's timestamps are useless as they wrongly show t1 instead of t.

**Comment by LittleVulpix** [comment]
@aaannndddyyy This is not a toxcore issue but a qtox issue. Qtox has a "faux offline" messaging whereby it will just deliver your messages whenever both you and your "target" are online at the same time. There is no implementation that would let the client use some extra metadata to see a timestamp - because this must be compatible with all of the other tox clients.

Therefore, this would need an implementation in core and then in all clients. It's almost a duplicate of: toxcore/870

**Comment by aaannndddyyy** [comment]
@LittleVulpix Yes, this is a core issue, and you yourself give the argument for that in your comment.
And not, it is only remotely related to #870. That one there is about a mehtod of storing and relaying messages, this one here is about timestamps.

**Comment by LittleVulpix** [comment]
@aaannndddyyy Well the way you described it, it's a qtox issue, because qtox does the faux offline messaging. I don't think another client does this (maybe antox?). Either way, the toxcore-side of this issue should be related to https://github.com/irungentoo/toxcore/issues/1418 in a way, because you essentially need to edit a sent message to give it a different timestamp than the one it gets assigned by your receiving client. The real trouble there is that you could run into bad time sync whereby the source client could for example have incorrectly set clock, and it would result in messages being forever in the past and/or forever in the future. It's very tricky to do this without a central server that's timesynced using NTP. I'm curious how this will be implemented.

**Comment by aaannndddyyy** [comment]
no, not related to 1418 at all.
It is core related and not qtox-related, because a lcient cannot know at what time a message was sent, if the client does not tell this.
It affects several clients.
